department-of-veterans-affairs/developer-portal
# Add skip navigation link

supports [API-171](https://vajira.max.gov/browse/API-171). adds a skip navigation link so that keyboard and screenreader users don't have to go through the whole site navigation every time they start on the site. design is loosely based on the skip nav from https://accessibility.digital.gov/.

see [WCAG Success Criterion 2.4.1: Bypass Blocks](https://www.w3.org/WAI/WCAG21/Understanding/bypass-blocks.html) for background info.

### Testing

for the skip navigation link:

* open the dev portal and tab once. you should see the focused skip navigation link in the top left. hit Enter. the site should have scrolled down to the `main` element, and if you tab again, focus should move to the first focusable element within the main page content.
* and that's pretty much it! try that from a few different pages to make sure it's robust, but that's it.
* the skip nav link should be visible if and only if it has focus (which has to come from tabbing, since you can't locate it with the mouse). 
* _screen reader:_ open a dev portal page with the screen reader on. navigate into the web content and use the right arrow to get to the first item in the page. it should be the skip nav link. press the link and navigate to the next item using the right arrow. it should be the first element in the main page content (side nav for pages with the side nav).

for the `PageContent` refactor:

* ensure that when you navigate to a different page, keyboard focus goes back to the main element. tabbing should move focus to the first focusable element in the page. for most pages, this is the Overview item in the side nav. shift-tabbing should move focus to the StatusPage link.
* ensure that if you are scrolled down within a page and navigate to a new page, the window scrolls to the top. a good test for this is the Support link in the beta banner at the bottom of the site.
